ttnn/cpp/ttnn/operations/data_movement/concat/concat.cpp

```diff
@@ -97,36 +110,60 @@ MassagedConcat build_untilize_rm_retilize_concat(uint8_t queue_id, const MemoryC
+                [=](const ttnn::Tensor& input_tensor) -> ttnn::Tensor {
                     TT_FATAL(
                         input_tensor.get_layout() == ttnn::TILE_LAYOUT,
                         "ttnn.concat: expected all input tensors to be in tile layout");
                     auto untilized_tensor = ttnn::untilize(input_tensor);
-                    // untilized, so now we have a padded rm tensor
+                    // untilized, so now we have a padded rm tensor. we slice to
+                    // remove the padding.
+                    std::vector<uint32_t> begins_vec(input_tensor.get_shape().rank(), 0);
+                    tt::stl::Span<const uint32_t> begins = begins_vec;
+                    tt::stl::Span<const uint32_t> ends = input_tensor.get_logical_shape().view();
+                    std::vector<uint32_t> steps_vec(input_tensor.get_shape().rank(), 1);
+                    tt::stl::Span<const uint32_t> steps = steps_vec;
+
+                    // we now perform a padding-oblivious slice to remove the
+                    // tile padding.
+                    // FIXME: change this to a legit slice call once
+                    // padding-oblivious entry point is uplifted to the slice
+                    // op.
+                    untilized_tensor = operation::run(
+                        SliceDeviceOperation{begins, ends, steps, output_memory_config},
+                        {untilized_tensor},
+                        {},
+                        {std::nullopt},
+                        queue_id)[0];
+
                     untilized_tensor = ttnn::reshape(
                         untilized_tensor,
-                        ttnn::Shape{
-                            input_tensor.get_logical_shape().view(), untilized_tensor.get_padded_shape().view()});
+                        ttnn::Shape{input_tensor.get_logical_shape().view(), input_tensor.get_logical_shape().view()});
                     return untilized_tensor;
                 });
             return std::make_tuple(itensors, dim, groups);
         },
-        .post_transform = [queue_id](const ttnn::Tensor& output) -> ttnn::Tensor {
+        .post_transform = [logical_output_shape, queue_id](const ttnn::Tensor& output) -> ttnn::Tensor {
             // now we have a rm tensor, so we need ensure its's padded to tile size and re-tilize it
             if (output.get_layout() != ttnn::TILE_LAYOUT) {
                 auto padded = pad_to_tile_vol(queue_id, output, 0.0f, true, output.memory_config());
                 concat_db_print(true, "[DEBUG] padded to tile layout, now tilizing.");
                 auto tilized =
                     ttnn::tilize_with_val_padding(padded, padded.get_legacy_shape(), 0.0f, output.memory_config());
                 concat_db_print(true, "[DEBUG] tilized");
-                return tilized;
+                // need to reshape tilized result to logical concat output shape
+                auto reshaped = ttnn::reshape(
```

Comment:
Makes me think we must change reshape to accept logical shape only (not in this PR)